### PR TITLE
refactor(components): [collapse] refactor

### PR DESCRIPTION
**Pull request by buqiyuan**
Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

File: packages/components/collapse/src/collapse.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import { buildProps, definePropType } from '@element-plus/utils'
+import { UPDATE_MODEL_EVENT, CHANGE_EVENT } from '@element-plus/constants'
+import type { ExtractPropTypes } from 'vue'
+import type Collapse from './collapse.vue'
+
+export type ModelValueType = string | number | Array<string | number>
```
Comment (sxzz) [inline packages/components/collapse/src/collapse.ts:6]: We can add a until type `Arrayable` to `packages/utils/typescript.ts`

```ts
export type Arrayable<T> = T[] | T
```

Comment (sxzz) [reply packages/components/collapse/src/collapse.ts:6]: And please rename it to `CollapseModelValue`

File: packages/components/collapse/src/collapse.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+export const emitChangeFn = (value: ModelValueType) =>
+  typeof value === 'number' || typeof value === 'string' || Array.isArray(value)
```
Comment (sxzz) [inline packages/components/collapse/src/collapse.ts:9]: Please use utils `isString`, `isNumber` from `@element-plus/utils`

**Review by sxzz — APPROVED** [review]

File: packages/components/collapse/src/collapse.ts (hunk 2 above)
Comment (buqiyuan) [reply packages/components/collapse/src/collapse.ts]: thanks~❤️，I also reviewed！

**Pull request merged**